### PR TITLE
Branch setcrs testing

### PR DESCRIPTION
**Pull request by saiutkarsh33**
- Created unit tests for set course command and its related classes

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/CustomAssertionError.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.commands;
+
+/**
+ * CustomAssertionError extends AssertionError to provide a more specific error indication
+ * for scenarios where a method is invoked inappropriately. It simplifies the process
+ * of throwing assertion errors with a default or custom message across the application.
+ */
+public class CustomAssertionError extends AssertionError {
```
Comment (tituschewxj) [inline src/test/java/seedu/address/logic/commands/CustomAssertionError.java:8]: **Unless** we have other custom assertions to add, consider changing the name `CustomAssertionError` to `InappropriateMethodCallAssertionError`, as it reduces ambiguities.

>[!IMPORTANT]
>You should change all instances of this name if you decide to change the name.

File: src/test/java/seedu/address/logic/commands/CustomAssertionError.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+     * Throws a CustomAssertionError with the default message.
+     * This static method provides a convenient way to throw an assertion error indicating
+     * that a certain operation or method call is invalid and should not have occurred.
+     * The default message is used for this error.
+     *
+     * @throws CustomAssertionError always, with the default error message.
+     */
+    public static void throwDefault() {
+        throw new CustomAssertionError();
+    }
```
Comment (tituschewxj) [inline src/test/java/seedu/address/logic/commands/CustomAssertionError.java:41]: If you are sticking with `CustomAssertionError` for the class name, we should change this method name to `throwInappropriateMethodCallAssertion`.

Then we should need to have another constructor and use the constant for this.
```suggestion
    public static void throwInappropriateMethodCallAssertion() {
        throw new CustomAssertionError(INAPPROPRIATE_METHOD_CALL_MESSAGE);
    }
```

File: src/test/java/seedu/address/logic/commands/CustomAssertionError.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package seedu.address.logic.commands;
+
+/**
+ * CustomAssertionError extends AssertionError to provide a more specific error indication
+ * for scenarios where a method is invoked inappropriately. It simplifies the process
+ * of throwing assertion errors with a default or custom message across the application.
+ */
+public class CustomAssertionError extends AssertionError {
+    // Static message for the error
+    private static final String DEFAULT_MESSAGE = "This method should not be called.";
```
Comment (tituschewxj) [inline src/test/java/seedu/address/logic/commands/CustomAssertionError.java:10]: Perhaps a better name would be:
```suggestion
    private static final String INAPPROPRIATE_METHOD_CALL_MESSAGE = "This method should not be called.";
```

**Review by tituschewxj — CHANGES_REQUESTED** [review]
Thanks for adding the tests! I have nits for the naming in `CustomAssertionError`, as the naming is quite ambiguous currently.

**Review by tituschewxj — APPROVED** [review]
LGTM! You don't have to do the changes in my note in this PR, since I don't want to hold back this PR from being merged. But if you do, then it's good too.

>[!NOTE]
>Instead of `InappropriateMethodCallAssertionError.throwDefault();`, you can just use `throw new InappropriateMethodCallAssertionError();`. And you can remove the `InappropriateMethodCallAssertionError#throwDefault` method, and the `InappropriateMethodCallAssertionError(String message)` constructor after doing this.

File: src/test/java/seedu/address/logic/commands/InappropriateMethodCallAssertionError.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+     * The default message is used for this error.
+     *
+     * @throws InappropriateMethodCallAssertionError always, with the default error message.
+     */
+    public static void throwDefault() {
+        throw new InappropriateMethodCallAssertionError();
+    }
+}
+
+
```
Comment (ForAeons) [inline src/test/java/seedu/address/logic/commands/InappropriateMethodCallAssertionError.java:44]: Can we move this file into commands/exceptions/

Comment (ForAeons) [inline src/test/java/seedu/address/logic/commands/InappropriateMethodCallAssertionError.java:44]: Can we move this class into commands/exception package

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 5]
```diff
@@ -14,27 +14,29 @@
...
 public class ParserUtilTest {
-    // TODO
     private static final String INVALID_NAME = "R@chel";
     private static final String INVALID_PHONE = "+651234";
     private static final String INVALID_ADDRESS = " ";
     private static final String INVALID_EMAIL = "example.com";
     private static final String INVALID_TAG = "#friend";
-
+    private static final String INVALID_COURSE = "CS";
     private static final String VALID_NAME = "Rachel Walker";
```
Comment (ForAeons) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:31]:
```suggestion
    private static final String INVALID_COURSE = "CS";
    
    private static final String VALID_NAME = "Rachel Walker";
```

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 6]
```diff
@@ -14,27 +14,29 @@
...
     private static final String VALID_PHONE = "123456";
     private static final String VALID_ADDRESS = "123 Main Street #0505";
     private static final String VALID_EMAIL = "rachel@example.com";
     private static final String VALID_TAG_1 = "friend";
     private static final String VALID_TAG_2 = "neighbour";
 
+    private static final String VALID_COURSE = "CS2101";
```
Comment (ForAeons) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:38]:
```suggestion
    private static final String VALID_COURSE = "CS2101";
```

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 7]
```diff
@@ -63,6 +66,19 @@ public void addressBookReadSave() throws Exception {
...
+         * Note: This is an integration test that verifies the StorageManager is properly wired to the
+         * {@link JsonCourseNameStorage} class.
+         * More extensive testing of UserPref saving/reading is done in {@link JsonStorageTest} class.
+         */
+        CourseName original = getTypicalCourseName();
+        storageManager.saveCourse(original);
+        ReadOnlyCourseName retrieved = storageManager.readCourse().get();
+        assertEquals(original, new CourseName(retrieved));
+    }
+
```
Comment (ForAeons) [inline src/test/java/seedu/address/storage/StorageManagerTest.java:83]:
```suggestion
```

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 8]
```diff
@@ -90,115 +90,124 @@ public void toStringMethod() {
...
-    // TODO
-    // Abstract away the assertion errors below
+
     /**
      * A default model stub that have all the methods failing.
      */
     private class ModelStub implements Model {
         @Override
         public void setUserPrefs(ReadOnlyUserPrefs userPrefs) {
-            throw new AssertionError("This method should not be called.");
```
Comment (ForAeons) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:101]: I agree with titus on this. I think is good to explicitly use the throw keyword

**Review by ForAeons — CHANGES_REQUESTED** [review]
Just some nits

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 9]
```diff
@@ -90,115 +90,124 @@ public void toStringMethod() {
...
+            InappropriateMethodCallAssertionError.throwDefault();
         }
```
Comment (ForAeons) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:101]: I agree with titus on this. I think is good to explicitly use the throw keyword

**Review by tituschewxj — APPROVED** [review]
LGTM!

**Review by ForAeons — APPROVED** [review]
Thanks a lot for making the changes! LGTM

**Pull request merged**